internal/perms/perms_test.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// System permissions-related code unit tests.
+package perms_test
+
+import (
+	"syscall"
+	"testing"
+
+	"github.com/googlecloudplatform/gcsfuse/internal/perms"
+	. "github.com/jacobsa/ogletest"
+)
+
+func TestPerms(t *testing.T) { RunTests(t) }
+
+////////////////////////////////////////////////////////////////////////
+// Boilerplate
+////////////////////////////////////////////////////////////////////////
+
+type PermsTest struct {
+}
+
+var _ SetUpInterface = &PermsTest{}
+var _ TearDownInterface = &PermsTest{}
+
+func init() { RegisterTestSuite(&PermsTest{}) }
+
+func (t *PermsTest) SetUp(ti *TestInfo) {
+}
+
+func (t *PermsTest) TearDown() {
+}
+
+////////////////////////////////////////////////////////////////////////
+// Tests
+////////////////////////////////////////////////////////////////////////
+
+func (t *PermsTest) Default() {
+	_, _, err := perms.MyUserAndGroup()
+	ExpectEq(err, nil)
+}
+
+func (t *PermsTest) ChangedUidGid() {
+	new_uid, new_gid := 111, 117
+	if err1, err2 := syscall.Setuid(new_uid), syscall.Setgid(new_gid); err1 == nil && err2 == nil {
```

Comment:
I dont get it. If setuid is throwing error why are writing this unit test?

Reply:
I had kept this test as a fallback. Will remove.